src/com/sap/piper/JenkinsUtils.groovy

```diff
@@ -19,6 +19,31 @@ static boolean hasTestFailures(build){
     return action && action.getFailCount() != 0
 }
 
+boolean addWarningsNGParser(String id, String name, String regex, String script, String example = ''){
+    def classLoader = this.getClass().getClassLoader()
+    def config = classLoader.loadClass('io.jenkins.plugins.analysis.warnings.groovy.ParserConfiguration', true)?.getInstance()
+
+    if(config.contains(id)){
+        return false
+    }else{
+        config.setParsers(
+            config.getParsers().plus(
+                classLoader.loadClass('io.jenkins.plugins.analysis.warnings.groovy.GroovyParser', true)?.newInstance(id, name, regex, script, example)
+            )
+        )
+        return true
+    }
+}
+
+@NonCPS
+static String getFullBuildLog(currentBuild) {
+    Reader reader = currentBuild.getRawBuild().getLogReader()
+    String logContent = org.apache.commons.io.IOUtils.toString(reader);
+    reader.close();
+    reader = null
+    return logContent
```

Comment:
Maybe this comment is a little bit paranoid: We might get a very long string, I remember back some time where that could get a memory issue, but maybe that is history in the meantime. Nevertheless would it be possible to stream the log rather than having it as a string? Basically we put it in another file.

Reply:
Step handled a job log with 80.000+ lines without an issue in an acceptable amount of time.